nucypher/blockchain/eth/actors.py

```diff
@@ -585,6 +594,83 @@ def encrypt_threshold_decryption_response(
             requester_public_key=requester_public_key,
         )
 
+
+    def _authorize_ritual(self, decryption_request: ThresholdDecryptionRequest):
+        # TODO: #3052 consider using the DKGStorage cache instead of the coordinator agent
+        # dkg_public_key = this_node.dkg_storage.get_public_key(decryption_request.ritual_id)
+        ritual = self.coordinator_agent.get_ritual(
+            decryption_request.ritual_id, with_participants=True
+        )
+
+        # enforces that the node is part of the ritual
+        participants = [p.provider for p in ritual.participants]
+        if self.checksum_address not in participants:
+            raise self.UnauthorizedRequest(
+                f"Node not part of ritual {decryption_request.ritual_id}",
+            )
+
+        # check that ritual not timed out
+        current_timestamp = self.coordinator_agent.blockchain.get_blocktime()
+        if current_timestamp > ritual.end_timestamp:
+            raise self.UnauthorizedRequest(
+                f"Ritual {decryption_request.ritual_id} is expired",
+            )
+
+    def _authorize_ciphertext(
+        self, decryption_request: ThresholdDecryptionRequest
+    ) -> None:
+        # check whether enrico is authorized
+        ciphertext_header = decryption_request.ciphertext_header
+        authorization = decryption_request.acp.authorization
+        if not self.coordinator_agent.is_encryption_authorized(
+            ritual_id=decryption_request.ritual_id,
+            evidence=authorization,
+            ciphertext_header=bytes(ciphertext_header),
+        ):
+            raise self.UnauthorizedRequest(
+                f"Encrypted data not authorized for ritual {decryption_request.ritual_id}",
+            )
+
+    def _authorize_decryption_request(
```

Comment:
Similar comment as above. Is this more like a check/validation than "authorize" action?